firefox-ios/Client/ApplicationHelper.swift

```diff
@@ -46,4 +47,18 @@ struct DefaultApplicationHelper: ApplicationHelper {
             open(url)
         }
     }
+
+    /// Closes the first tab that matches the url passed in
+    /// This is most likely from other clients connected to the same
+    /// account requesting to close the tab on this device
+    ///
+    /// - Parameters:
+    ///   - url: the URL requested to be closed
+    func closeTab(_ url: URL) async {
+        let windowManager = AppContainer.shared.resolve() as WindowManager
+        for tabManager in windowManager.allWindowTabManagers() where
+            await tabManager.removeTabByUrl(url) {
+                break
+            }
```

Comment:
This seems like it could be cleaned up slightly but LMK if I'm misunderstanding. Possibly something like:

```
windowManager.allWindowTabManagers().forEach {
    await _ = $0.removeTabByUrl(url)
}
```

Reply:
ah my goal was to exit out of the loop once we find a matching url so we don't unnecessarily iterate through all the tabs and end up removing more than one instance of the url (if there were duplicates).

Reply:
Ah ok sorry I misunderstood the intent 👍